modules/service/src/main/scala/lucuma/odb/service/CalibrationsService.scala

```diff
@@ -3,39 +3,82 @@
 import lucuma.odb.util.Codecs.*
 import lucuma.odb.util.GmosCodecs.*
-import org.typelevel.log4cats.Logger
+import lucuma.refined.*
 import skunk.Query
+import skunk.Transaction
 import skunk.syntax.all.*
 
 trait CalibrationsService[F[_]] {
   def recalculateCalibrations(
     pid: Program.Id
-  )(using L: Logger[F]): F[Unit]
+  )(using Transaction[F]): F[Unit]
 
 }
 
 object CalibrationsService {
-  def instantiate[F[_]: FlatMap](using Services[F]): CalibrationsService[F] =
+  def instantiate[F[_]: Monad](using Services[F]): CalibrationsService[F] =
     new CalibrationsService[F] {
-      def recalculateCalibrations(pid: Program.Id)(using L: Logger[F]): F[Unit] =
+      private val CalibrationsGroupName: NonEmptyString = "Calibrations".refined
+
+      private def calibrationsGroup(pid: Program.Id, size: Int)(using Transaction[F]): F[Option[Group.Id]] =
+        if (size > 0) {
+          groupService.selectGroups(pid).flatMap {
+            case GroupTree.Root(_, c) =>
+              val existing = c.collectFirst {
+                case GroupTree.Branch(gid, _, _, _, Some(CalibrationsGroupName), _, _, _, true) => gid
+              }
+              // Create a system group for calibrations if it does not exist
+              existing match {
+                case Some(gid) => gid.some.pure[F]
+                case None      =>
+                  groupService.createGroup(
+                      input = CreateGroupInput(
+                        programId = pid.some,
+                        proposalReference = none,
+                        programReference = none,
+                        SET = GroupPropertiesInput.Create(
+                          name = CalibrationsGroupName.some,
+                          description = CalibrationsGroupName.some,
+                          minimumRequired = none,
+                          ordered = false,
+                          minimumInterval = none,
+                          maximumInterval = none,
+                          parentGroupId = none,
+                          parentGroupIndex = none,
+                          existence = Existence.Present
+                        )
+                      ),
+                    system = true
+                  ).map(_.toOption)
+              }
+            case _ => none.pure[F]
+          }
+        } else none.pure[F]
+
+      def recalculateCalibrations(pid: Program.Id)(using Transaction[F]): F[Unit] =
         for {
-          gnls   <- session.execute(Statements.SelectGmosNorthLongSlitConfigurations)(pid)
-          gsls   <- session.execute(Statements.SelectGmosSouthLongSlitConfigurations)(pid)
-          groups <- groupService.selectGroups(pid)
-          _      <- Logger[F].info(groups.toString)
+          gnls <- session.execute(Statements.SelectGmosNorthLongSlitConfigurations)(pid)
+          gsls <- session.execute(Statements.SelectGmosSouthLongSlitConfigurations)(pid)
+          _    <-  calibrationsGroup(pid, gnls.size + gsls.size)
```

Comment:
These unique configurations will be used to create the actual calibrations in the future correct?

Reply:
Yes, that's correct we create a new obs for each unique configuration